src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -0,0 +1,144 @@
+    (compojure/GET "/file_bucket_file/*" request
+      (master-request-handler (map-accept-s-pson-to-binary request)))
+    ;; TODO: file_bucket_file request PUTs from Puppet agents currently use a
+    ;; Content-Type of 'text/plain', which, per HTTP specification, would imply
+    ;; a default character encoding of ISO-8859-1 or US-ASCII be used to decode
+    ;; the data.  This would be incorrect to do in this case, however, because
+    ;; the actual payload is "binary".  Coercing this to
+    ;; "application/octet-stream" for now as this is synonymous with "binary".
+    ;; This should be removed when/if Puppet agents start using an appropriate
+    ;; Content-Type to describe the input payload - see PUP-3812 for the core
+    ;; Puppet work and SERVER-294 for the related Puppet Server work that
+    ;; would be done.
+    (compojure/PUT "/file_bucket_file/*" request
+      (master-request-handler
+        (-> request
+          (map-accept-s-pson-to-binary)
+          (assoc :content-type "application/octet-stream") ; deprecated in ring >= 1.2
+          (assoc-in [:headers "content-type"] "application/octet-stream")))) ; ring >= 1.2 spec
+    (compojure/HEAD "/file_bucket_file/*" request
+      (master-request-handler (map-accept-s-pson-to-binary request)))
+    (compojure/GET "/catalog/*" request (master-request-handler request))
+    (compojure/POST "/catalog/*" request (master-request-handler request))
+    (compojure/PUT "/report/*" request (master-request-handler request))
+    (compojure/GET "/resource_type/*" request (master-request-handler request))
+    (compojure/GET "/resource_types/*" request (master-request-handler request))
+    (compojure/GET "/status/*" request (master-request-handler request))
+
+    ;; TODO: when we get rid of the legacy dashboard after 3.4, we should remove
+    ;; this endpoint as well.  It makes more sense for this type of query to be
+    ;; directed to PuppetDB.
+    (compojure/GET "/facts_search/*" request (master-request-handler request))
+
+    ;; legacy CA routes
+    (compojure/ANY "/certificate_status/*" request (ca-request-handler request))
+    (compojure/ANY "/certificate_statuses/*" request (ca-request-handler request))
+    (compojure/GET "/certificate/*" request (ca-request-handler request))
+    (compojure/GET "/certificate_revocation_list/*" request (ca-request-handler request))
+    (compojure/GET "/certificate_request/*" request (ca-request-handler request))
+    (compojure/GET "/certificate_requests/*" request (ca-request-handler request))))
+
+(defn legacy-routes
+  "Creates all of the compojure routes for the master."
+  [master-request-handler ca-request-handler]
+  (compojure/routes
+    (compojure/context "/v2.0" request
+                       (v2_0-routes master-request-handler))
+    (compojure/context "/:environment" [environment]
+                       (legacy-master-routes master-request-handler ca-request-handler))
+    (route/not-found "Not Found")))
```

Comment:
The presence of a `not-found` route here makes me nervous because we know we have to mount this handler at "" on the server.  I'd worry that if this were running in a stack that had non-master / non-CA routes registered that the "not-found" route here could swallow up any requests to those other routes.  I think it would be safer to just not include that here, like we didn't for the implementations of these routes in the Puppet Server 1.x implementation.